support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -96,31 +100,39 @@ class ZuoraSubscriptionCreator(
         .withEventualLogging("subscribe")
       _ <- updateRedemptionCodeIfApplicable(state.paymentMethod, corporateCodeStatusUpdater)
         .withEventualLogging("update redemption code")
-    } yield HandlerResult(getEmailState(state, account, sub, paymentOrRedemptionData, maybeGiftPurchase), info)
+    } yield HandlerResult(new NextState(state, account, sub).getEmailState(paymentOrRedemptionData, maybeDigitalSubscriptionGiftCreationDetails), info)
   }
 
-  private def addGiftPurchaseDetails(billingPeriod: BillingPeriod, giftRecipient: GiftRecipient, today: () => LocalDate) =
+  private def addDigitalSubscriptionGiftCreationDetails(billingPeriod: BillingPeriod, giftRecipient: GiftRecipient, today: () => LocalDate) =
     giftRecipient match {
       case digitalSubscriptionGiftRecipient: DigitalSubscriptionGiftRecipient =>
         val giftCode = giftCodeGeneratorService.generateCode(billingPeriod)
           .withLogging("Generated code for Digital Subscription gift")
         val lastRedemptionDate = today().plusMonths(GiftCodeValidator.expirationTimeInMonths).minusDays(1)
-        DigitalSubscriptionGiftPurchase(digitalSubscriptionGiftRecipient, giftCode, lastRedemptionDate)
-      case weeklyGiftRecipient: WeeklyGiftRecipient =>
-        WeeklyGiftPurchase(weeklyGiftRecipient)
+        Some(DigitalSubscriptionGiftCreationDetails(digitalSubscriptionGiftRecipient, giftCode, lastRedemptionDate))
+      case _ =>
+        None
     }
 
 }
 
 object ZuoraSubscriptionCreator {
 
+  case class DigitalSubscriptionGiftCreationDetails(
```

Comment:
What do you think about calling this `DigitalSubscriptionGiftPurchaseDetails` I think it would be good to standardise on calling the two types of gift checkout 'Purchase' & 'Redemption'. Creation doesn't seem very specific to me.

Reply:
will do, yeah I agree standardisation is good here

Reply:
fixed this now.